File: Stage20/C/Problem15649.c
#include <stdio.h>

int N, M;
int arr[9];
int visited[9];

void dfs(int depth) {
    if (depth == M) {
        for (int i = 0; i < M; i++) printf("%d ", arr[i]);
        printf("\n");
        return;
    }

    for (int i = 1; i <= N; i++) {
        if (!visited[i]) {
            visited[i] = 1;
            arr[depth] = i;
            dfs(depth + 1);
            visited[i] = 0;
        }
    }
}

int main(void) {
    scanf("%d %d", &N, &M);

    dfs(0);
    return 0;
}

File: Stage20/Python/Problem15649.py
def dfs(depth):
    if depth == M:
        print(*arr)
        return

    for i in range(1, N + 1):
        if not visited[i]:
            visited[i] = True
            arr.append(i)
            dfs(depth + 1)
            visited[i] = False
            arr.pop()

def main():
    global N, M, arr, visited
    N, M = map(int, input().split())
    arr = []
    visited = [False] * (N + 1)
    dfs(0)
    
if __name__ == "__main__":
    main()